src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -881,6 +889,311 @@ public void RestoreOriginalParentTest()
+        [InlineData($"W3C,              Unknown,            Unknown,            NonW3C,     true,                   false,              W3C,                    W3C")]
+        [InlineData($"Hierarchical,     Unknown,            Unknown,            W3C,        true,                   false,              Hierarchical,           Hierarchical")]
+        [InlineData($"W3C,              Unknown,            Unknown,            null,       false,                  false,              W3C,                    W3C")]
+        [InlineData($"Hierarchical,     Unknown,            Unknown,            null,       false,                  false,              Hierarchical,           Hierarchical")]
+
+        [InlineData($"Unknown,          W3C,                Unknown,            null,       true,                   false,              W3C,                    Default")]
+        [InlineData($"Hierarchical,     W3C,                Unknown,            null,       true,                   false,              W3C,                    Hierarchical")]
+        [InlineData($"Unknown,          W3C,                Hierarchical,       null,       true,                   false,              W3C,                    Hierarchical")]
+        [InlineData($"Hierarchical,     W3C,                Hierarchical,       null,       true,                   false,              W3C,                    Hierarchical")]
+        [InlineData($"Hierarchical,     W3C,                Hierarchical,       NonW3C,     true,                   false,              W3C,                    Hierarchical")]
+
+        [InlineData($"Unknown,          Hierarchical,       Unknown,            null,       true,                   false,              Hierarchical,           Default")]
+        [InlineData($"W3C,              Hierarchical,       Unknown,            null,       true,                   false,              Hierarchical,           W3C")]
+        [InlineData($"Unknown,          Hierarchical,       W3C,                null,       true,                   false,              Hierarchical,           W3C")]
+        [InlineData($"W3C,              Hierarchical,       Hierarchical,       null,       true,                   false,              Hierarchical,           W3C")]
+        [InlineData($"W3C,              Hierarchical,       Hierarchical,       W3C,        true,                   false,              Hierarchical,           W3C")]
+        [InlineData($"Unknown,          Hierarchical,       Unknown,            null,       false,                  false,              Hierarchical,           W3C")]
+        [InlineData($"Unknown,          W3C,                Unknown,            null,       false,                  false,              W3C,                    W3C")]
+
+        [InlineData($"Unknown,          W3C,                Unknown,            null,       true,                   true,               W3C,                    Default")]
```

Comment:
We should have the test case where a non-W3C string id gets passed as the parentId, the child does get W3C format, and the ActivityCreationOptions\<ActivityContext>.TraceId property does get viewed inside the Sample callback. That case wasn't previously calling Sample and we need to ensure that now it does work correctly.

Reply:
You mean when the listener is not providing `SampleUsingParentId`. right?

Reply:
Right!